src/ReactiveUI.SourceGenerator.Tests/TestHelper.cs

```diff
@@ -150,29 +167,41 @@ public GeneratorDriver RunGeneratorAndCheck<T>(
             throw new InvalidOperationException("Must have a valid compiler instance.");
         }
 
+        // Add this assembly as a reference.
+        IEnumerable<MetadataReference> thisAssembly = [MetadataReference.CreateFromFile(typeof(TestHelper).Assembly.Location)];
+
         // Collect required assembly references.
         var assemblies = new HashSet<MetadataReference>(
             Basic.Reference.Assemblies.Net80.References.All
+            .Concat(thisAssembly)
             .Concat(_eventCompiler.Modules.Select(x => MetadataReference.CreateFromFile(x.PEFile!.FileName)))
             .Concat(_eventCompiler.ReferencedModules.Select(x => MetadataReference.CreateFromFile(x.PEFile!.FileName)))
             .Concat(_eventCompiler.NeededModules.Select(x => MetadataReference.CreateFromFile(x.PEFile!.FileName))));
 
+        var syntaxTree = CSharpSyntaxTree.ParseText(code, CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp10));
+
         // Create a compilation with the provided source code.
         var compilation = CSharpCompilation.Create(
             "TestProject",
-            [CSharpSyntaxTree.ParseText(code)],
+            [syntaxTree],
             assemblies,
             new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, deterministic: true));
 
-        // Optionally validate diagnostics before running the generator.
+        // Validate diagnostics before running the generator.
+        var prediagnostics = compilation.GetDiagnostics()
+                .Where(d => d.Severity > DiagnosticSeverity.Warning)
+                .ToList();
         if (ignoreConditional is not null)
         {
-            var diagnostics = compilation.GetDiagnostics().ToList();
-            diagnostics.Where(x => !ignoreConditional(x)).Should().BeEmpty();
+            prediagnostics.Where(x => !ignoreConditional(x)).Should().BeEmpty();
```

Comment:
prediagnostics should fail most of the time, since the source generated code you rely on isn't there.

Reply:
Having Diagnostics fail the test helped in the diagnosis of the Test code, and enabled visibility of what needed to be masked